RecoTracker/TkSeedingLayers/interface/SeedingHitSet.h

```diff
@@ -45,34 +46,34 @@ class SeedingHitSet {
   }
 
   SeedingHitSet(std::vector<ConstRecHitPointer> &hits) {
+    initContainer();
     if (hits.size() > seedinghits::kMaxSeedSize)
       edm::LogWarning("SeedingHitSet") << "More than " << seedinghits::kMaxSeedSize
                                        << " hits give in input. Taking only the first " << seedinghits::kMaxSeedSize;
```

Comment:
I'm not sure this qualifies for a warning, it's at best a LogInfo. 

A more controlled behavior would make more sense to me though.
I suggest to add a flag (e.g. `bool allowTruncation = false`) and silently truncate if truncation is OK and assert otherwise.